src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -209,16 +210,18 @@ private BoundAttribute BindAttributeCore(AttributeSyntax node, NamedTypeSymbol a
                 {
                     Error(diagnostics, ErrorCode.ERR_AttributeCtorInParameter, node, attributeConstructor.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat));
                 }
+
+                BindDefaultArguments(node, attributeConstructor.Parameters, analyzedArguments.ConstructorArguments.Arguments, analyzedArguments.ConstructorArguments.RefKinds, ref argsToParamsOpt, out defaultArguments, expanded, enableCallerInfo: true, diagnostics);
```

Comment:
This change feels scary a little bit.

Reply:
The reason I called `BindDefaultArguments` is to populate `ConstructorArguments` with non-explicit arguments for optional parameters. This is because `DeriveArguments` won't work without those arguments being populated.

@RikkiGibson @333fred It looks like `BindDefaultArguments` was specifically written for invocations, not attributes, and I think it's incorrect to call it here. Do you have any suggestions?

Reply:
It seems attributes do argument reordering at binding-time as well since there are no side effects from the arguments. That's interesting.